src/Symfony/Component/Console/Input/InputDefinition.php

```diff
@@ -246,6 +248,14 @@ public function addOption(InputOption $option)
                 $this->shortcuts[$shortcut] = $option->getName();
             }
         }
+
+        if ($option->isNegatable()) {
+            $negatedName = 'no-'.$option->getName();
+            if (isset($this->options[$negatedName])) {
+                throw new LogicException(sprintf('An option named "%s" already exists.', $negatedName));
```

Comment:
this validation misses a case: when adding the explicit `no-foo` option *after* the negated option.

Reply:
good catch. Fixed by #39723